robot/impl/local_robot.go

```diff
@@ -1145,6 +1165,7 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 	if err != nil {
 		allErrs = multierr.Combine(allErrs, err)
 	}
+	r.applyLocalModuleVersions(newConfig)
```

Comment:
Thanks for pointing me to this! I get how this works now. I'd appreciate if we can document this step here with:
```
For hot reloadable local tgz modules, we create synthetic versions for package
management. The `localRobot` keeps track of which module name is running which version. Here,
we copy the synthetic version from the `localRobot` into the appropriate `config.Module`
object inside the `cfg.Modules` slice. Thus, when a local tgz module is reloaded, the
viam-server can unpackage it into a known fresh directory.
```

Feel free to use that as-is or put that into your own words.

Reply:
done in b58e891c0